src/core/adapters/SimplePoolAdapter.ts

```diff
@@ -257,33 +268,44 @@ export abstract class SimplePoolAdapter implements IProtocolAdapter {
       eventResults.map(async (transferEvent) => {
         const {
           blockNumber,
-          args: { value },
+          args: { value: protocolTokenMovementValueRaw },
         } = transferEvent
 
         const protocolTokenPrice = await this.getPricePerShare({
           blockNumber,
           protocolTokenAddress,
         })
 
-        const pricePerShareRaw =
-          protocolTokenPrice.tokens?.[0]?.pricePerShareRaw
-        if (!pricePerShareRaw) {
-          throw new Error('No price for events at this time')
-        }
-
-        const movementValueRaw = BigInt(value.toString()) * pricePerShareRaw
         return {
           underlyingTokensMovement: underlyingTokens.reduce(
-            (accummulator, currentToken) => {
+            (accumulator, currentToken) => {
+              const currentTokenPrice = protocolTokenPrice.tokens?.find(
+                (price) => price.address === currentToken.address,
```

Comment:
Good catch here. We were picking always the first token, but that only works if there's only one underlying.